### PR TITLE
Rewrite the public instances list

### PR DESCRIPTION
**Pull request by TheFrenchGhosty**
Rendered version: https://github.com/TheFrenchGhosty/documentation/blob/instances-list-rewrite/Public-Instances.md

This PR is a rewrite of the public instances list from scratch.

- Instances list is now in a table, making things a LOT easier to read
- The plain text version of the table is readable (disable word wrap)
- A lot of information have been added to ease the management (mainly internally) but also so that users are aware of who runs X instance
- Prerequisites have been rewritten better, and new one have been added

---

It is "incomplete" in the sense that it's missing the link to some privacy policies, and the owners of the Tor onion, but it's enough for https://github.com/iv-org/instances-api to be adapted to work with it.

**It mustn't be merged before https://github.com/iv-org/instances-api is ready to support it.** (@syeopite seems to plan to rewrite its parser)

---

Took me around to do ~2 hours, it was a lot more time consuming that I expected.

---

Note:

The wording of some categories have to be discussed (mainly "DDos Protection / MITM").

### CODE REVIEW

**Comment by syeopite** [comment]
https://invidious.048596.xyz is ran by @tenpura-shrimp  See #6

**Comment by TheFrenchGhosty** [comment]
@syeopite You're right, I got confused by this commit: https://github.com/iv-org/documentation/commit/a6fdd0eb07ec0fce433ea9b3c9652f3c45118dba / https://github.com/iv-org/documentation/pull/39

File: Public-Instances.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+| [grwp24hodrefzvjjuccrkw3mjq4tzhaaq32amf33dzpmuxe7ilepcmad.onion](http://grwp24hodrefzvjjuccrkw3mjq4tzhaaq32amf33dzpmuxe7ilepcmad.onion)  | 🇺🇸 USA          | ?                                                                                             |                                             | ?                                                      |                                                                                                                                                      |
+| [hpniueoejy4opn7bc4ftgazyqjoeqwlvh2uiku2xqku6zpoa4bf5ruid.onion](http://hpniueoejy4opn7bc4ftgazyqjoeqwlvh2uiku2xqku6zpoa4bf5ruid.onion)  | 🇺🇸 USA          | [invidious.kavin.rocks](https://invidious.kavin.rocks)                                        |                                             | [@FireMasterK](https://github.com/FireMasterK)         |                                                                                                                                                      |
```
Comment (FireMasterK) [inline Public-Instances.md:34]: This should be `invidious-us.kavin.rocks`

File: Public-Instances.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+| [c7hqkpkpemu6e7emz5b4vyz7idjgdvgaaa3dyimmeojqbgpea3xqjoid.onion](http://c7hqkpkpemu6e7emz5b4vyz7idjgdvgaaa3dyimmeojqbgpea3xqjoid.onion)  | ?               | ?                                                                                             |                                             | ?                                                      |                                                                                                                                                      |
+| [w6ijuptxiku4xpnnaetxvnkc5vqcdu7mgns2u77qefoixi63vbvnpnqd.onion](http://w6ijuptxiku4xpnnaetxvnkc5vqcdu7mgns2u77qefoixi63vbvnpnqd.onion)  | ?               | ?                                                                                             |                                             | ?                                                      |                                                                                                                                                      |
```
Comment (FireMasterK) [inline Public-Instances.md:31]: This is my other onion instance, located in India.

**Comment by FireMasterK** [comment]
Should something like "Instances using a modified source code, are required to publish their source code before they can be added to the list, and must timely publish any later modification in a timely manner." be added to prevent another #72 from happening?

File: Public-Instances.md (hunk 1 above)
Comment (TheFrenchGhosty) [reply Public-Instances.md]: Oops typo, thanks.

File: Public-Instances.md (hunk 2 above)
Comment (TheFrenchGhosty) [reply Public-Instances.md]: Good to know, thanks :)

**Comment by TheFrenchGhosty** [comment]
@FireMasterK 

> Should something like "Instances using a modified source code, are required to publish their source code before they can be added to the list, and must timely publish any later modification in a timely manner." be added to prevent another #72 from happening?

Done, thanks :)

**Comment by syeopite** [comment]
I think we should add an uptime requirement. Any instance that is down most of the time should not be added to the official list imo.

**Comment by TheFrenchGhosty** [comment]
@syeopite 

> I think we should add an uptime requirement. Any instance that is down most of the time should not be added to the official list imo.

This is a good idea, should how much should we require? 75%?

**Comment by unixfox** [comment]
> @syeopite
> 
> > I think we should add an uptime requirement. Any instance that is down most of the time should not be added to the official list imo.
> 
> This is a good idea, should how much should we require? 75%?

At the very least 90% because it's still like almost 3 days per months of downtime:
image

**Comment by TheFrenchGhosty** [comment]
> > @syeopite
> > > I think we should add an uptime requirement. Any instance that is down most of the time should not be added to the official list imo.
> > 
> > 
> > This is a good idea, should how much should we require? 75%?
> 
> At the very least 90% because it's still like almost 3 days per months of downtime:
> image
> Source: https://uptime.is/

Yeah, you're right. 90% is the way to go I guess.

**Comment by syeopite** [comment]
This PR can get merged soon. I've finished the parser rewrite for the instances-api a few days ago. See iv-org/instances-api#30

**Comment by syeopite** [comment]
Owner for https://notyoutube.org is wrong. The actual owner is dejalavidavolar. See #58. But due to #82 it should be removed

**Pull request closed without merging**